classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.php

```diff
@@ -0,0 +1,203 @@
+        Schema::table('submission_files', function (Blueprint $table) {
+            $table->dropForeign('submission_files_file_id_foreign');
+            $table->foreign('file_id')->references('file_id')->on('files')->onDelete('cascade');
+        });
+
+        // Does have the foreign key reference but not the CASCADE
+        Schema::table('submission_file_revisions', function (Blueprint $table) {
+            $table->dropForeign('submission_file_revisions_submission_file_id_foreign');
+            $table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files')->onDelete('cascade');
+
+            $table->dropForeign('submission_file_revisions_file_id_foreign');
+            $table->foreign('file_id')->references('file_id')->on('files')->onDelete('cascade');
+        });
+
+        // Does not have the foreign key reference
+        Schema::table('submission_file_settings', function (Blueprint $table) {
+            $table->bigInteger('submission_file_id')->nullable(false)->unsigned()->change();
+            $table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files')->onDelete('cascade');
+        });
+
+        // Does have the foreign key reference but not the CASCADE
+        Schema::table('review_files', function (Blueprint $table) {
+            $table->dropForeign('review_files_submission_file_id_foreign');
+            $table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files')->onDelete('cascade');
+        });
+
+        // Does have the foreign key reference but not the CASCADE
+        Schema::table('review_round_files', function (Blueprint $table) {
+            $table->dropForeign('review_round_files_submission_file_id_foreign');
+            $table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files')->onDelete('cascade');
+        });
+
+        // Does not have the foreign key reference
+        Schema::table('query_participants', function (Blueprint $table) {
+            $table->foreign('query_id')->references('query_id')->on('queries')->onDelete('cascade');
+            $table->foreign('user_id')->references('user_id')->on('users')->onDelete('cascade');
+        });
+
+        $orphanedIds = DB::table('notes AS n')
+            ->leftJoin('queries AS q', 'n.assoc_id', '=', 'q.query_id')
+            ->where('n.assoc_type', '=', self::ASSOC_TYPE_QUERY)
+            ->whereNull('q.query_id')
+            ->pluck('n.note_id', 'n.assoc_id');
+
+        foreach ($orphanedIds as $neQueryId => $noteId) {
+            $notesFileRows = DB::table('submission_files as sf')
+                ->join('files as f', 'sf.file_id', '=', 'f.file_id')
+                ->where('sf.assoc_type', '=', self::ASSOC_TYPE_NOTE)
+                ->where('sf.assoc_id', '=', $noteId)
+                ->where('sf.file_stage', '=', self::SUBMISSION_FILE_QUERY)
```

Comment:
You're about to delete the note, so regardless of the `sf.file_stage`, any related `submission_files` entries should be deleted too. You can remove this check on `sf.file_stage`.

Reply:
Correct. Fixed